### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by recurser**
## Add tests

[Clubhouse Story](https://app.clubhouse.io/shuttlerock/story/69519)

## How to test the PR

If tests pass, it works.

## Deployment Notes

None.

### CODE REVIEW

File: lib/trailer/storage/cloud_watch.rb [hunk 1]
```diff
@@ -49,6 +39,16 @@ def flush
...
+      # Queues the given hash for writing to CloudWatch.
+      #
+      # @param data [Hash] A key-value hash of trace data to write to storage.
+      def write(data)
+        messages << {
+          timestamp: (Time.now.utc.to_f.round(3) * 1000).to_i,
+          message:   data&.to_json,
+        }.compact
+      end
+
```
Comment (recurser) [inline lib/trailer/storage/cloud_watch.rb:53]: Just putting methods in alphabetical order - no logic changes.

**Review by VaguelyElectric — APPROVED** [review]

File: lib/trailer/concern.rb [hunk 2]
```diff
@@ -27,7 +28,7 @@ def trace_event(event, resource = nil, **tags, &block) # rubocop:disable Metrics
           end
         elsif resource.respond_to?(:to_h)
           # This handles other types of data, such as GraphQL input objects.
-          resource.to_h.stringify_keys.each do |key, value|
+          resource.to_h.transform_keys(&:to_s).each do |key, value|
             tags[key] ||= value if key.to_s.match?(Trailer.config.auto_tag_fields) || Trailer.config.tag_fields.include?(key)
```
Comment (avokhmin) [inline lib/trailer/concern.rb]:
```suggestion
            tags[key] ||= value if key.match?(Trailer.config.auto_tag_fields) || Trailer.config.tag_fields.include?(key)
```

Comment (avokhmin) [reply lib/trailer/concern.rb]: PS: maybe here better to use a symbols too?

File: lib/trailer/storage/cloud_watch.rb [hunk 3]
```diff
@@ -49,6 +39,16 @@ def flush
         retry
       end
 
+      # Queues the given hash for writing to CloudWatch.
+      #
+      # @param data [Hash] A key-value hash of trace data to write to storage.
+      def write(data)
+        messages << {
+          timestamp: (Time.now.utc.to_f.round(3) * 1000).to_i,
+          message:   data&.to_json,
```
Comment (avokhmin) [inline lib/trailer/storage/cloud_watch.rb:50]: PS: maybe we can skip the messages without data?

**Review by avokhmin — COMMENTED** [review]
👍

File: lib/trailer/storage/cloud_watch.rb (hunk 3 above)
Comment (recurser) [reply lib/trailer/storage/cloud_watch.rb:50]: Good idea, I missed it 👍

File: lib/trailer/concern.rb (hunk 2 above)
Comment (recurser) [reply lib/trailer/concern.rb]: Good catch 👍

File: lib/trailer/storage/cloud_watch.rb [hunk 4]
```diff
@@ -49,6 +39,18 @@ def flush
         retry
       end
 
+      # Queues the given hash for writing to CloudWatch.
+      #
+      # @param data [Hash] A key-value hash of trace data to write to storage.
+      def write(data)
+        return if data.empty?
```
Comment (avokhmin) [inline lib/trailer/storage/cloud_watch.rb:46]: maybe:

```
      def write(data)
        return if data.nil? || data.empty?

        messages << {
          timestamp: (Time.now.utc.to_f.round(3) * 1000).to_i,
          message:   data.to_json,
        }
      end
```

**Review by avokhmin — COMMENTED** [review]
👍

**Pull request merged**